app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/ReviewRegistrationController.scala

```diff
@@ -111,13 +107,16 @@ class ReviewRegistrationController @Inject() (
    */
   def submit(): Action[AnyContent] =
     (authenticate andThen journeyAction).async { implicit request =>
-      val referenceId = s"PPT12345678${Random.nextInt(1000000)}"
-      successSubmissionCounter.inc()
-      markRegistrationAsCompleted().map {
-        case Right(reg) =>
-          auditor.registrationSubmitted(reg)
-          Redirect(routes.ConfirmationController.displayPage())
-            .flashing(Flash(Map(FlashKeys.referenceId -> referenceId)))
+      markRegistrationAsCompleted().flatMap {
+        case Right(updatedRegistration) =>
+          subscriptionsConnector.submitSubscription(getSafeId(updatedRegistration),
+                                                    updatedRegistration
+          ).map { response =>
+            successSubmissionCounter.inc()
+            auditor.registrationSubmitted(updatedRegistration)
+            Redirect(routes.ConfirmationController.displayPage())
+              .flashing(Flash(Map(FlashKeys.referenceId -> response.pptReference)))
+          }
         case Left(error) =>
```

Comment:
not sure if this case is possible now, we're throwing error on submission error in the connector. Maybe it is better to respond with Either[error, success] from subscriptionConnector and then increment the failure submission counter on error handling. What do you think? Unless I'm missing something, please validate I'm only looking at diffs. Whatever we do we just want to have a way of incrementing this failed submission counter (other option could be to move that counters success/failure to the connector)

Reply:
yes a very good point , I will move that to the connector as part of the next ticket